compiler/rustc_expand/src/expand.rs

```diff
@@ -793,6 +798,36 @@ impl<'a, 'b> MacroExpander<'a, 'b> {
                 }
                 _ => unreachable!(),
             },
+            InvocationKind::GlobDelegation { item } => {
+                let AssocItemKind::DelegationMac(deleg) = &item.kind else { unreachable!() };
```

Comment:
can't we change `Invocation::GlobDelegation` to hold `Box<ast::DelegationMac>` and the item's `Span` to eliminate this "unwrap"?

Reply:
Expansion uses several other parts of the item (attributes, visibility, span).
Seems simpler to just carry the whole item around.